### ISSUE TITLE
Be consistent in type hinting named parameters

### ISSUE THREAD

**Issue by giovannt0** [google/cloud-forensics-utils#171]
Ideally we would have `def foo(bar: Optional[Type] = None) -> ReturnType` for named parameters which default to `None` instead of  `def foo(bar: Type = None) -> ReturnType`. As of now we have both across the codebase.

Note: we want to use `Optional` where it makes sense from `mypy` POV. `def foo(bar: Optional[str] = 'default-string') -> ReturnType` does not make sense because `Optional` essentially means `Union[type, None]`. In this case this would mean `Union[str, None]` and unless we're specifically allowing / expecting that `bar` may be `None`, `Optional` shouldn't be used here.

**Comment by hacktobeer** [comment]
If a parameter can have a None/'' value it's Optional, else it's not Optional, correct? Named parameters are Optional.

**Comment by giovannt0** [comment]
Yes yes, `mypy` accepts both forms of type hinting but I find it cleaner to explicitly use the `Optional` keyword in the method's signature (and most methods are currently using it, so it's more about using it everywhere or not use it at all).

**Comment by hacktobeer** [comment]
but we should only use Optional when the parameter is optional, right?

**Comment by giovannt0** [comment]
Yes! Anything of the form `func(param = value)` would be type hinted as `func(param: Optional[Type] = value)`

**Comment by tomchop** [comment]
> If a parameter can have a None/'' value it's Optional, else it's not Optional, correct? Named parameters are Optional.

to clarify: as far as mypy is concerned, `Optional` parameters should be used only for keywords arguments that have no default value (i.e. that default to `None`). Any keyword argument that declares a default value is implicitly optional for mypy, and its type defaults to the type of the declared default value.

TLDR: We want to use `Optional` only when the parameter is optional and we expect that parameter to be `None` if it is not explicitly passed by the caller.
